internal/controller/postgrescluster/instance.go

```diff
@@ -1759,3 +1804,51 @@ func (r *Reconciler) prepareForUpgrade(ctx context.Context,
 	}
 	return nil
 }
+
+// +kubebuilder:rbac:groups=policy,resources=poddisruptionbudgets,verbs=create;patch;list;delete
+
+// reconcileInstanceSetPodDisruptionBudget creates a PDB for an instance set. A
+// PDB will be created when the minAvailable is determined to be greater than 0.
+// MinAvailable can be defined in the spec or a default value will be set based
+// on the number of replicas in the instance set.
+func (r *Reconciler) reconcileInstanceSetPodDisruptionBudget(
+	ctx context.Context,
+	cluster *v1beta1.PostgresCluster,
+	spec *v1beta1.PostgresInstanceSetSpec,
+) error {
+	if spec.Replicas == nil {
+		// Replicas should always have a value because of defaults in the spec
+		return errors.New("Replicas should be defined")
+	}
+	minAvailable := getMinAvailable(spec.MinAvailable, *spec.Replicas)
+
+	meta := naming.InstanceSet(cluster, spec)
+	meta.Labels = naming.Merge(cluster.Spec.Metadata.GetLabelsOrNil(),
+		spec.Metadata.GetLabelsOrNil(),
+		map[string]string{
+			naming.LabelCluster:     cluster.Name,
+			naming.LabelInstanceSet: spec.Name,
+		})
+	meta.Annotations = naming.Merge(cluster.Spec.Metadata.GetAnnotationsOrNil(),
+		spec.Metadata.GetAnnotationsOrNil())
+
+	selector := naming.ClusterInstanceSet(cluster.Name, spec.Name)
+	pdb, err := r.generatePodDisruptionBudget(cluster, meta, minAvailable, selector)
+
+	// If 'minAvailable' is set to '0', we will not reconcile the PDB. If one
+	// already exists, we will remove it.
+	if err == nil &&
+		(minAvailable.Type == 0 && minAvailable.IntVal == 0) ||
+		(minAvailable.Type == 1 && (minAvailable.StrVal == "0%" || minAvailable.StrVal == "00%")) {
```

Comment:
Does an "" register as anything with intOrString?

Reply:
`""` is an invalid value when the spec is created. We aren't doing validation on our spec so it is possible to pass it through and end up with an error when you try to parse the  `IntOrString` or apply the PDB spec. So far we have been punting on validation but I do need to add in a check for the error. I can also make a note to valid in the future/with a webhook.